### ISSUE TITLE
pmdarima imports from `sklearn.externals`, which raises a warning

### ISSUE THREAD

**Issue by telamonian** [tgsmith61591/pmdarima#147]
`sklearn.externals.six` and `sklearn.externals.joblib` have been deprecated in the latest version of `sklearn`. Unfortunately, various code in pmdarima imports directly from these deprecated packages. Thus, when I run:

```python
from pmdarima.arima import auto_arima
```

I get a bunch of garbage warnings printed to my stdout:

```
/usr/local/lib/python3.7/site-packages/sklearn/externals/six.py:31: DeprecationWarning: The module is deprecated in version 0.21 and will be removed in version 0.23 since we've dropped support for Python 2.7. Please rely on the official version of six (https://pypi.org/project/six/).
  "(https://pypi.org/project/six/).", DeprecationWarning)
/usr/local/lib/python3.7/site-packages/sklearn/externals/joblib/__init__.py:15: DeprecationWarning: sklearn.externals.joblib is deprecated in 0.21 and will be removed in 0.23. Please import this functionality directly from joblib, which can be installed with: pip install joblib. If this warning is raised when loading pickled models, you may need to re-serialize those models with scikit-learn 0.21+.
  warnings.warn(msg, category=DeprecationWarning)
```

Like the warnings say, it would be better if pmdarima depended upon the actual `six` and `joblib` packages, instead of the ones currently (but not for much longer) bundled with `sklearn`.

**Comment by ClaudeCoulombe** [comment]
I've got the problem so I've patched each imports in:

pyramid/utils/__init__.py
pyramid/arima/seasonality.py
pyramid/arima/stationarity.py
pyramid/utils/array.py

#    from sklearn.externals import six
 import six

and in

pyramid/arima/auto.py

# from sklearn.externals.joblib import Parallel, delayed
from joblib import Parallel, delayed

**Comment by ClaudeCoulombe** [comment]
Ok, finally I've found 

pmdarima version: 1.6.0

and get it using 

`sudo pip3 install pmdarima`
